core/sdk-core/src/main/java/software/amazon/awssdk/core/metrics/CoreMetric.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.metrics;
+
+import java.time.Duration;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.metrics.MetricCategory;
+import software.amazon.awssdk.metrics.SdkMetric;
+
+@SdkPublicApi
+public final class CoreMetric {
+    /**
+     * The unique ID for the service. This is present for all API call metrics.
+     */
+    public static final SdkMetric<String> SERVICE_ID = metric("ServiceName", String.class);
+
+    /**
+     * The name of the service operation being invoked. This is present for all
+     * API call metrics.
+     */
+    public static final SdkMetric<String> OPERATION_NAME = metric("OperationName", String.class);
+
+    /**
+     * The duration of the API call. This includes all call attempts made.
+     */
+    public static final SdkMetric<Duration> API_CALL_DURATION = metric("ApiCallDuration", Duration.class);
+
+    /**
+     * The duration of time taken to marshall the SDK request to an HTTP
+     * request.
+     */
+    public static final SdkMetric<Duration> MARSHALLING_DURATION = metric("MarshallingDuration", Duration.class);
+
+    /**
+     * The duration of time taken to fetch signing credentials for the request.
+     */
+    public static final SdkMetric<Duration> CREDENTIALS_FETCH_DURATION = metric("CredentialsFetchDuration", Duration.class);
+
+    /**
+     * The duration fo time taken to sign the HTTP request.
+     */
+    public static final SdkMetric<Duration> SIGNING_DURATION = metric("SigningDuration", Duration.class);
+
+    /**
+     * The total time take to send a HTTP request and receive the response.
+     */
+    public static final SdkMetric<Duration> HTTP_REQUEST_ROUND_TRIP_TIME = metric("HttpRequestRoundTripTime", Duration.class);
```

Comment:
Do we have plans to include `HttpClientSendRequestTime` and `HttpClientReceiveResponseTime` like we have in v1?

Reply:
I thought about this but I didn't see the value in it compared to just the duration time. WDYT? I can go either way but I can't think of a situation where it would be useful

Reply:
I think granular metrics will provide more insight. For example, for a slow request with higher HttpClientSendRequestTime, we could know that the slowness is in the request, probably something related to establishing a new connection.

Reply:
How would you know that a request is slow based on HttpClientSendRequestTime though? You'd need some other time to compare it to. In your example, I would rather we tracked the time taken to open a connection as its own metric

Reply:
Yeah, `HttpClientSendRequestTime` itself doesn't really make much sense. I meant by looking at all three metrics. `HttpClientSendRequestTime`, `HttpClientReceiveResponseTime` and `HttpRequestRoundTripTime`. For some http clients like http url connection client, it might not be possible to track the time taken to open a connection.